lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -59,13 +59,15 @@ def collection_ancestor? collections
     end
 
     def ancestor? collection
-      return true if collection == self
+      return true if (( !collection.id.nil? && !self.id.nil? ) && ( collection.id == self.id ))
+      return true if (( collection.id.nil? || self.id.nil? ) && ( collection == self ))
```

Comment:
@jcoyne It looks like core.rb is doing a similar comparison.  I don't know how Ruby optimizes code.  In other languages, the way this is written, the full object comparison will happen first and the shorter checks on class and id will only happen if comparison_object.equal?(self) returns false because evaluation happens left to right.  
```
    def ==(comparison_object)
         comparison_object.equal?(self) ||
           (comparison_object.instance_of?(self.class) &&
             comparison_object.id == id &&
             !comparison_object.new_record?)
    end
```